FirebaseAI/Sources/Types/Public/Live/LiveServerContent.swift

```diff
@@ -0,0 +1,83 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Incremental server update generated by the model in response to client
+/// messages.
+///
+/// Content is generated as quickly as possible, and not in realtime. Clients
+/// may choose to buffer and play it out in realtime.
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *)
+@available(watchOS, unavailable)
+public struct LiveServerContent: Sendable {
+  let serverContent: BidiGenerateContentServerContent
+
+  /// The content that the model has generated as part of the current
+  /// conversation with the user.
+  ///
+  /// This can be ``null`` if the message signifies something else (such
+  /// as the turn ending).
+  public var modelTurn: ModelContent? { serverContent.modelTurn }
+
+  /// The model has finished sending data in the current turn.
+  ///
+  /// Generation will only start in response to additional client messages.
+  ///
+  /// Can be set alongside ``content``, indicating that the ``content`` is
+  /// the last in the turn.
+  public var isTurnComplete: Bool? { serverContent.turnComplete }
+
+  /// The model was interrupted by a client message while generating data.
+  ///
+  /// If the client is playing out the content in realtime, this is a
+  /// good signal to stop and empty the current queue.
+  public var wasInterrupted: Bool? { serverContent.interrupted }
+
+  /// The model has finished _generating_ data for the current turn.
+  ///
+  /// For realtime playback, there will be a delay between when the model finishes generating
+  /// content and the client has finished playing back the generated content. ``generationComplete``
+  /// indicates that the model is done generating data, while ``isturnComplete`` indicates the model
+  /// is waiting for additional client messages. Sending a message during this delay may cause a
+  /// ``wasInterrupted`` message to be sent.
+  ///
+  ///  Note that if the model ``wasInterrupted``, this will not be set. The model will go from
+  /// ``wasInterrupted`` -> ``turnComplete``.
+  public var isGenerationComplete: Bool? { serverContent.generationComplete }
```

Comment:
Should we make these booleans non-optional? Is there any value to the developer to know which of the 3 states (`nil`, `false`, `true`)? The `false` values are likely always going to be `nil` when false, at least with Vertex AI, since they don't serialize default values in the JSON.

```
  public var isTurnComplete: Bool { serverContent.turnComplete ?? false }
  public var wasInterrupted: Bool { serverContent.interrupted ?? false }
  public var isGenerationComplete: Bool { serverContent.generationComplete ?? false }
```

cc: @ncooke3